zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
+
+    Each individual asset's data is stored as a bcolz table with a column for
+    each pricing field: (open, high, low, close, volume)
+
+    The open, high, low, and close columns are integers which are 1000 times
+    the quoted price, so that the data can represented and stored as an
+    np.uint32, supporting market prices quoted up to the thousands place.
+
+    volume is a np.uint32 with no mutation of the tens place.
+
+    The 'index' for each individual asset are a repeating period of minutes of
+    length `minutes_per_day` starting from each market open.
+    The file format does not account for half-days.
+    e.g.:
+    2016-01-19 14:31
+    2016-01-19 14:32
+    ...
+    2016-01-19 20:59
+    2016-01-19 21:00
+    2016-01-20 14:31
+    2016-01-20 14:32
+    ...
+    2016-01-20 20:59
+    2016-01-20 21:00
+
+    All assets are written with a common 'index', sharing a common first
+    trading day. Assets that do not begin trading until after the first trading
+    day will have zeros for all pricing data up and until data is traded.
+
+    'index' is in quotations, because bcolz does not provide an index. The
+    format allows index-like behavior by writing each minute's data into the
+    corresponding position of the enumeration of the aforementioned datetime
+    index.
+
+    The datetimes which correspond to each position are written in the metadata
+    as integer nanoseconds since the epoch into the `minute_index` key.
+    """
+    def __init__(self,
+                 first_trading_day,
+                 rootdir,
+                 market_opens,
+                 minutes_per_day=MINUTES_PER_DAY,
+                 ohlc_ratio=OHLC_RATIO,
+                 expectedlen=DEFAULT_EXPECTEDLEN):
+        """
+        Parameters:
+        -----------
+
+        first_trading_day : datetime-like
+            The first trading day in the data set.
+        
+        rootdir : string
+            Path to the root directory into which to write the metadata and
+            bcolz subdirectories.
+
+        market_opens : pd.Series
+            The market opens used as a starting point for each periodic span of
+            minutes in the index.
+
+        minutes_per_day : int
+            The number of minutes per each period. Defaults to 390, the mode
+            of minutes in NYSE trading days.
+
+        ohlc_ratio : int
+            The ratio by which to multiply the pricing data to convert the floats
+            from floats to an integer to fit within the np.uint32.
+
+            The default is 1000 to support pricing data which comes in to the
+            thousands place.
+
+        expectedlen : int
+            The expected length of the dataset, used when creating the initial
+            bcolz ctable.
+
+            If the expectedlen is not used, the chunksize and corresponding
+            compression ratios are not ideal.
+
+            Defaults to supporting 15 years of market data.
```

Comment:
This is 15 years of market data for NYSE equities, right?  Might want to specify that here.